### PR TITLE
[pytket-pyquil] Upgraded to pyQuil v3.0

### PR DESCRIPTION
**Pull request by lmondada**

### CODE REVIEW

File: modules/pytket-pyquil/pytket/extensions/pyquil/pyquil_convert.py [hunk 1]
```diff
@@ -299,39 +312,84 @@ def process_characterisation(qc: QuantumComputer) -> dict:
...
+    # T1s and T2s are currently left empty
+    t1_times_dict: dict = {}
+    t2_times_dict: dict = {}
+
+    for q in isa.qubits.values():
+        node = Node(q.id)
+        for g in q.gates:
+            if g.fidelity is None:
+                g.fidelity = 1.0
+            print(g)
```
Comment (cqc-alec) [inline modules/pytket-pyquil/pytket/extensions/pyquil/pyquil_convert.py:344]: Remove this line.

File: modules/pytket-pyquil/pytket/extensions/pyquil/pyquil_convert.py [hunk 2]
```diff
@@ -299,39 +312,84 @@ def process_characterisation(qc: QuantumComputer) -> dict:
...
+            if isinstance(g, GateInfo) and g.operator in str_to_gate_1qb:
+                angle = _get_angle_type(g.parameters[0])
+                if angle is not None:
+                    try:
+                        optype = str_to_gate_1qb[g.operator][angle]
+                    except KeyError:
```
Comment (cqc-alec) [inline modules/pytket-pyquil/pytket/extensions/pyquil/pyquil_convert.py:350]: In this case I think `optype` won't be set to anything so the below code will fail; is there anything we can do in that case or should we just raise an error here?

File: modules/pytket-projectq/docs/changelog.rst [hunk 3]
```diff
@@ -1,6 +1,11 @@
 Changelog
 ~~~~~~~~~
 
+0.12.0 (XXXX 2021)
+------------------
+
+* Updated projectq version requirement to 0.7.
```
Comment (cqc-alec) [inline modules/pytket-projectq/docs/changelog.rst:7]: Why is this here?

**Review by cqc-alec — CHANGES_REQUESTED** [review]

File: modules/pytket-projectq/docs/changelog.rst (hunk 3 above)
Comment (cqc-alec) [reply modules/pytket-projectq/docs/changelog.rst:7]: Ah I see, it is fixing up the previous PR.

File: modules/pytket-pyquil/pytket/extensions/pyquil/pyquil_convert.py (hunk 2 above)
Comment (lmondada) [reply modules/pytket-pyquil/pytket/extensions/pyquil/pyquil_convert.py:350]: good spot, it needs to skip the rest in that case.

File: modules/pytket-projectq/docs/changelog.rst (hunk 3 above)
Comment (lmondada) [reply modules/pytket-projectq/docs/changelog.rst:7]: This wasn't added to a previous PR (my fault). Or should I put this somewhere else?

File: modules/pytket-pyquil/docs/changelog.rst [hunk 4]
```diff
@@ -1,6 +1,11 @@
 Changelog
 ~~~~~~~~~
 
+0.15.0 (XXXX 2021)
```
[the remark this answers is not in the document]
Comment (lmondada) [reply modules/pytket-pyquil/docs/changelog.rst:4]: done

File: modules/pytket-pyquil/pytket/extensions/pyquil/pyquil_convert.py (hunk 1 above)
Comment (lmondada) [reply modules/pytket-pyquil/pytket/extensions/pyquil/pyquil_convert.py]: done

**Review by cqc-alec — APPROVED** [review]
